sdk/src/executor/process/utils.rs

```diff
@@ -32,25 +36,113 @@ macro_rules! craft_child_process {
 }
 
 #[allow(unused_results)]
-pub(crate) fn create_stream(mut child: Child) -> Lines<BufReader<tokio::process::ChildStdout>> {
-    // Create stream to read output
-    let stdout = child
-        .stdout
-        .take()
-        .expect("Failed to take stdout handle from child...");
-    let reader = BufReader::new(stdout).lines();
+pub(crate) fn create_stream(mut child: Child) -> broadcast::Receiver<String> {
+    let (tx, rx) = broadcast::channel(1000);
+    let stdout = child.stdout.take().expect("Failed to take stdout");
+    let stderr = child.stderr.take().expect("Failed to take stderr");
+
+    let mut stdout_reader = BufReader::new(stdout).lines();
+    let mut stderr_reader = BufReader::new(stderr).lines();
+
+    const CHILD_PROCESS_TIMEOUT: Duration = Duration::from_millis(10000);
+    const MAX_CONSECUTIVE_NONE: usize = 5;
 
-    // Run in Tokio runtime to ensure it stays alive
     tokio::spawn(async move {
+        let mut stdout_none_count = 0;
+        let mut stderr_none_count = 0;
+
+        loop {
+            tokio::select! {
+                result = stdout_reader.next_line() => {
+                    crate::error!("READ TO STDOUT: {:?}", result);
+                    match result {
+                        Ok(Some(line)) => {
+                            stdout_none_count = 0;
+                            if !line.is_empty() && tx.send(format!("stdout: {}", line)).is_err() {
+                                    break;
+                            }
+                        }
+                        Ok(None) => {
+                            stdout_none_count += 1;
+                            if stdout_none_count >= MAX_CONSECUTIVE_NONE {
+                                crate::info!("Reached maximum consecutive None values for stdout");
+                                break;
+                            }
+                        }
+                        Err(e) => {
+                            crate::error!("Error reading from stdout: {}", e);
+                            break;
+                        }
+                    }
+                }
+                result = stderr_reader.next_line() => {
+                    crate::error!("READ TO STDERR: {:?}", result);
```

Comment:
These logs and the processing logic look repetitive to above, can we reuse it with a closure? i.e. the program flow from stdout to stderr looks identical, is it not?